base/random.jl

```diff
@@ -1408,14 +1408,25 @@ end
 Base.show(io::IO, u::UUID) = write(io, Base.repr(u))
 
 # return a random string (often useful for temporary filenames/dirnames)
+
 let b = UInt8['0':'9';'A':'Z';'a':'z']
     global randstring
-    randstring(r::AbstractRNG, n::Int) = String(b[rand(r, 1:length(b), n)])
-    randstring(r::AbstractRNG) = randstring(r,8)
-    randstring(n::Int) = randstring(GLOBAL_RNG, n)
-    randstring() = randstring(GLOBAL_RNG)
+    randstring(r::AbstractRNG, chars::AbstractArray{<:Union{UInt8,Char}}=b, n::Int=8) =
```

Comment:
The choice of  `Char` and `UInt8` is because the `String` constructor accepts arrays of those types, and I thougth that `String(v::Vector{UInt8})` was taking ownership of `v` (according to the docs) so that it would be efficient, but I just did few tests and it seems that this constructor actually copies the data from the vector. Anyway, `randstring` is still about twice as fast when `chars` is `UInt8` data rather than `Char`.

(And I also find the default length surprisingly arbitrary, not sure what to do about it though).